framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -318,6 +336,23 @@ protected BaseTypeVisitor(BaseTypeChecker checker, Factory typeFactory) {
                 checker.hasOption("conservativeUninferredTypeArguments");
     }
 
+    /** Initialize AnnoToTargetLocations, which is used for declared typeuse locations lookup. */
```

Comment:
I would move this method further down in this class. Can you think of a nicer name? Again, qualifiers instead of annotations.
Maybe create the map instead of modifying it?

Reply:
how about `initQualAllowedLocations`?